controller/src/beerocks/master/db/db.cpp

```diff
@@ -2904,27 +2904,116 @@ bool db::update_client_persistent_db(const sMacAddr &mac)
     return true;
 }
 
-std::unordered_map<std::string, std::unordered_map<std::string, std::string>>
-db::load_persistent_db_clients()
+bool db::load_persistent_db_clients()
 {
     // if persistent db is disabled
     if (!config.persistent_db) {
         LOG(ERROR) << "persistent db is disabled";
-        return {};
+        return false;
     }
 
     std::unordered_map<std::string, std::unordered_map<std::string, std::string>> clients;
     if (!bpl::db_get_entries_by_type(type_to_string(beerocks::eType::TYPE_CLIENT), clients)) {
         LOG(ERROR) << "failed to get all clients from persistent DB";
-        return {};
+        return false;
     }
 
     if (clients.empty()) {
         LOG(DEBUG) << "persistent DB doesn't exist (or empty) or doesn't contain clients";
-        return {};
+        return false;
+    }
+
+    auto type_client_str                       = db::type_to_string(beerocks::eType::TYPE_CLIENT);
+    static const std::string invalid_timestamp = "invalid_str";
+
+    //validate clients data from persistent db
+    for (auto &client : clients) {
+        const auto &client_entry = client.first;
```

Comment:
So it is a mac, only formatted like this? `aa_bb_cc_dd_ee_ff` ?
Please check is it possible to use colons on the UCI if the phrase it wrapped with quotes os something.
Anyway, I don't think it is a good idea to just leave it like this, this is confusing already and will look ugly/strange on the log.
I think that on the load function you should convert whatever format you use on UCI, to `sMacAddr`.

Reply:
again - please differ between BPL-context - which is referred to as entry - to the DB/nodes context which we refer to as client-mac
and to your question - no - the UCI doesn't support the colons in any form (as section name, as the value of param, aka. option, it does)

I intentionally used the entry and not the transformation from entry to mac to allow easy debugging in anyone edited the UCI and didn't provide an appropriate entry name - you will easily find the "culprit" since you have the invalid entry-name.

this actually raises another case where the entry name is invalid - I will add this verification as part of the refactoring of this API implementation.